### PR TITLE
More clim-fig improvements

### PR DESCRIPTION
**Pull request by scymtym**
* Naively show final appearance while drawing a figure

* Add a gadget for choosing the line dashes style

* Disable/enable commands as appropriate (in addition to the corresponding buttons)

### CODE REVIEW

File: Examples/clim-fig.lisp [hunk 1]
```diff
@@ -325,25 +334,25 @@
...
-                                (declare (ignore x))
-                                (com-redo)))
-   (clear :push-button
-          :label "Clear"
-          :active nil
-          :activate-callback #'(lambda (x)
-                                 (declare (ignore x))
-                                 (com-clear)))
-   (status :label-pane :label "CLIM Fig"))
+   (com-undo :push-button ; NOTE-COMMAND{EN,DIS}ABLED relies on the pane name
```
[the remark this answers is not in the document]
Comment (nsrahmad) [reply Examples/clim-fig.lisp]: Here is one recommendation, restore the pane names as "undo", "redo" and "clear". Then generate notification methods as below:
```lisp
(macrolet ((define-notification-functions-for-command (command-name pane-name)
	     `(progn
		(defmethod note-command-enabled (frame-manager (frame clim-fig)
						 (command-name (eql ',command-name)))
		  (activate-gadget (find-pane-named frame ',pane-name))
		  (call-next-method))
		(defmethod note-command-disabled (frame-manager (frame clim-fig)
						  (command-name (eql ',command-name)))
		  (deactivate-gadget (find-pane-named frame ',pane-name))
		  (call-next-method)))))
  (define-notification-functions-for-command com-undo undo)
  (define-notification-functions-for-command com-redo redo)
  (define-notification-functions-for-command com-clear clear))
```

[the remark this answers is not in the document]
Comment (nsrahmad) [reply Examples/clim-fig.lisp]: Here is another one, not idiomatic CL but very easy for beginners to understand (I prefer this one).
```lisp
(defmethod note-command-enabled :after (frame-manager (frame clim-fig) command-name)
  (alexandria:switch (command-name)
    ('com-undo  (activate-gadget (find-pane-named frame 'undo)))
    ('com-redo  (activate-gadget (find-pane-named frame 'redo)))
    ('com-clear (activate-gadget (find-pane-named frame 'clear)))))

(defmethod note-command-disabled :after (frame-manager (frame clim-fig) command-name)
  (alexandria:switch (command-name)
    ('com-undo  (deactivate-gadget (find-pane-named frame 'undo)))
    ('com-redo  (deactivate-gadget (find-pane-named frame 'redo)))
    ('com-clear (deactivate-gadget (find-pane-named frame 'clear)))))
```

[the remark this answers is not in the document]
Comment (scymtym) [reply Examples/clim-fig.lisp]: I went with something very similar to your final suggestions.

**Comment by nsrahmad** [comment]
Looks good now, Thanks.

**Pull request merged**